finality-aleph/src/party/manager/mod.rs

```diff
@@ -350,11 +351,14 @@ where
         };
 
         let last_block_of_previous_session = session_boundaries.first_block().saturating_sub(1);
-        let last_block_of_previous_session_hash = self
-            .client
-            .block_hash(last_block_of_previous_session)
-            .expect("Previous session ended, the block should be present")
-            .expect("Previous session ended, we should have the hash.");
+        let last_block_of_previous_session_header = self
+            .header_backend
+            .header_of_finalized_at(last_block_of_previous_session)
+            .expect("Previous session ended, the block should be present.")
+            .expect("Previous session ended, the block should be finalized.");
+
+        let last_block_of_previous_session_hash =
+            UnverifiedHeader::id(&last_block_of_previous_session_header).hash();
```

Comment:
This is necessary because of the multiple traits that the header satisfies, right? If so, please add a comment here about replacing this with a normal dot call after the other traits are no longer present.

Reply:
Hmm, I could've written type equations differently, adding `H: Header` parameter and separating it from `HeaderT` (we'd need that parameter anyway in the future). Solved and now even the comment is not necessary